travis-script.sh

```diff
@@ -49,95 +49,85 @@ case $GHCVER in
 esac
 
 # ---------------------------------------------------------------------
+# Timing
+# ---------------------------------------------------------------------
+
+timed() {
+    echo "$1"
+    start_time=`date +%s`
+    $2
+    end_time=`date +%s`
+    echo "$1" took $(expr $end_time - $start_time) seconds.
+    echo "----"
+}
+
+# ---------------------------------------------------------------------
 # Bootstrap cabal, to verify bootstrap.sh script works.
 # ---------------------------------------------------------------------
 
-OLD_CWD=$PWD
+bootstrap() {
+    OLD_CWD=$PWD
```

Comment:
Might it be a better idea to use subshells to ensure the preservation of CWD?

(I mean, the build will certainly fail if it isn't restored properly, but...)